bluemira/builders/EUDEMO/maintenance.py

```diff
@@ -0,0 +1,257 @@
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+
+"""
+Some crude EU-DEMO remote maintenance considerations
+"""
+
+import numpy as np
+
+from bluemira.base.constants import EPS
+from bluemira.base.look_and_feel import bluemira_warn
+from bluemira.geometry.constants import VERY_BIG
+from bluemira.geometry.face import BluemiraFace
+from bluemira.geometry.plane import BluemiraPlane
+from bluemira.geometry.tools import boolean_cut, make_polygon, slice_shape
+from bluemira.utilities.opt_problems import (
+    OptimisationConstraint,
+    OptimisationObjective,
+    OptimisationProblem,
+)
+from bluemira.utilities.optimiser import Optimiser, approx_derivative
+
+
+def _get_inner_cut_point(breeding_blanket_xz, r_inner_cut):
+    """
+    Get the inner cut point of the breeding blanket geometry.
+    """
+    cut_plane = BluemiraPlane.from_3_points(
+        [r_inner_cut, 0, 0], [r_inner_cut, 0, 1], [r_inner_cut, 1, 1]
+    )
+    # Get the first intersection with the vertical inner cut plane
+    intersections = slice_shape(breeding_blanket_xz.boundary[0], cut_plane)
+    intersections = intersections[intersections[:, -1] > 0.0]
+    intersection = sorted(intersections, key=lambda x: x[-1])[0]
+    return intersection
+
+
+class UpperPortOP(OptimisationProblem):
+    """
+    Reduced model optimisation problem for the vertical upper port size optimisation.
+
+    Parameters
+    ----------
+    optimiser: Optimiser
+        Optimiser object to use when solving this problem
+    breeding_blanket_xz: BluemiraFace
+        Unsegmented breeding blanket x-z geometry
+    """
+
+    def __init__(self, params, optimiser: Optimiser, breeding_blanket_xz: BluemiraFace):
+
+        objective = OptimisationObjective(self.minimise_port_size, f_objective_args={})
+
+        box = breeding_blanket_xz.bounding_box
+        r_ib_min = box.x_min
+        r_ob_max = box.x_max
+        c_rm = params.c_rm.value
+        R_0 = params.R_0.value
+        bb_min_angle = 90 - params.bb_min_angle.value
+        tk_bb_ib = params.tk_bb_ib.value
+        tk_bb_ob = params.tk_bb_ob.value
+
+        constraints = [
+            OptimisationConstraint(
+                self.constrain_blanket_cut,
+                f_constraint_args={
+                    "bb": breeding_blanket_xz,
+                    "c_rm": c_rm,
+                    "r_ib_min": r_ib_min,
+                    "r_ob_max": r_ob_max,
+                },
+                tolerance=1e-6 * np.ones(3),
```

Comment:
I'm not clear where this 3 comes from, I guess it's the number of constraints? Maybe another one to extract into a variable?